### ISSUE TITLE
feat: Send push notifications when patch bundles are updated

### ISSUE THREAD

**Issue by KobeW50** [ReVanced/revanced-manager#2189]
### Feature description

Add an option to the ReVanced Manager settings that allows users to choose how often to check for updates to the user's patch bundles. There should be an option to never check.

The following screenshot is from [Bunny Manager](https://github.com/pyoncord/BunnyManager):
![SmartSelect_20240903_161714_Bunny Manager](https://github.com/user-attachments/assets/3883e500-29e7-4130-9750-dee6d47af137)

When an update is found, a push notification should be delivered to the user, informing the user which patch bundle has been updated.

### Motivation

Currently, users need to open the Manager to see if patches have been updated. This can easily be forgotten by users.

Push notifications for updates is a useful utility for many users, and it will promote people to keep their patched apps up-to-date.

### Acknowledgements

- [X] I have checked all open and closed feature requests and this is not a duplicate.
- [X] I have chosen an appropriate title.
- [X] The feature request is only related to ReVanced Manager.

**Comment by Ushie** [comment]
Why would users be interested in the patch bundles updating? Unless major fixes have been made, it's not intended for users to constantly check up on the patch bundles

**Comment by KobeW50** [comment]
Many people prefer to keep their software up-to-date. Not just for major fixes, but also minor bug fixes, as well as newly implemented features.

**Comment by KobeW50** [comment]
That's fine. Also it'sworth clarifying that the setting is only for checking in the background. There should always be a check when the user opens the app

**Comment by Brosssh** [comment]
This can probably be closed as the bundle auto update feature exists. No notification is needed if the manager already does the refresh in the background (imo).

**Comment by PalmDevs** [comment]
@Ushie If this were to be implemented, how do you plan on implementing it? I know that Android allows apps to periodically fetch data every set interval. Perhaps we can do that?

**Comment by Ushie** [comment]
Yeah but I'm thinking about the necessity of this, there's no particular reason to update patches unnecessarily, besides you might be a user of ReVanced Twitter for example and not care about reddit, yet nothing says that ReVanced Twitter is going to be updated in the next 20 releases
